src/server/ServerHandler.java

```diff
@@ -98,6 +99,9 @@ public Response handleRequest(Request request) {
 
         // Use switch-case to decide among the two request types
         switch (request.getType()) {
+            case REGISTER:
+                User user = gson.fromJson((String) request.getData(), User.class);
```

Comment:
Don't need the cast to String, getData() already returns a string
```suggestion
                User user = gson.fromJson(request.getData(), User.class);
```